### PR TITLE
Reject CREATE with a duplicate id

### PR DESCRIPTION
**Pull request by bajtos**
Modify the memory connector to reject requests to create a record with a duplicate id.

Add a unit-test to verify this behaviour across all connectors.

Connect to https://github.com/strongloop/loopback/issues/986

/to @raymondfeng please review

### CODE REVIEW

File: lib/connectors/memory.js [hunk 1]
```diff
@@ -210,6 +210,11 @@ Memory.prototype.create = function create(model, data, callback) {
   if(!this.collection(model)) {
     this.collection(model, {});
   }
+
+  if (this.collection(model)[id]) {
+    return callback(new Error('Duplicate entry for ' + model + '.' + idName));
```
Comment (raymondfeng) [inline lib/connectors/memory.js]: We should make the callback inside `process.nextTick` so that the async expectation is guaranteed.

Comment (bajtos) [inline lib/connectors/memory.js]: Good point, fixed.

**Comment by bajtos** [comment]
LGTY now?

**Comment by raymondfeng** [comment]
LGTM

**Pull request merged**